src/dask_awkward/lib/io/io.py

```diff
@@ -674,3 +675,49 @@ def bytes_reading_ingredients(
         sample_bytes = sample_bytes[:rfind]
 
     return out, sample_bytes
+
+
+class FromTextFn:
+    def __init__(self):
+        pass
+
+    def __call__(self, ingredients: tuple) -> ak.Array:
+        (fs, path, compression, offsets, length, delimiter) = ingredients
+
+        with fs.open(path, compression=compression) as f:
+            if offsets == 0 and length is None:
+                bytestring = f.read()
+            else:
+                bytestring = read_block(f, offsets, length, delimiter)
+
+            buffer = np.frombuffer(bytestring, dtype=np.uint8)
+            array = ak.from_numpy(buffer)
+            array = ak.unflatten(array, len(array))
+            array = ak.enforce_type(array, "string")
+            array_split = ak.str.split_pattern(array, "\n")
+            lines = array_split[0]
+            return lines
+
+
+def from_text(source, blocksize, delimiter, storage_options: dict | None = None):
+    from fsspec.core import get_fs_token_paths
+
+    fs, token, paths = get_fs_token_paths(source, storage_options=storage_options or {})
+
+    bytes_ingredients, sample_bytes = bytes_reading_ingredients(
+        fs,
+        paths,
+        None,
+        delimiter,
+        False,
+        blocksize,
+        "128 KiB",
```

Comment:
= sample size? It's a reasonable value, but user might need to change it. We don't actually need it at all, since we know the form of the output is array-of-string (if delimiter is None) or array-of-list-of-string (otherwise). Do we allow for delimiter=None?

Reply:
Yea it's the sample size; definitely planning on making it user definable! On delimiter- right now I'm strictly passing in b"\n" (temporary just to get the ball rolling), how to handle delimiters (a sensible default and what to supprt in general) was something I had in mind to discuss

Reply:
Actually we make no use of it, so the value should be None/False (whatever it takes not to read it)

Reply:
`delimiter=b"\n"` is a reasonable default

Reply:
something I'm perhaps over complicating but I'm getting caught up wrapping my head around the difference between the bytes reading delimiter and then the "awkward-level" delimiter. At the bytes reading delimiter we use `b"\n"` as a default; but then at the awkward level I currently have hardcoded in
`ak.str.split_pattern(array, "\n")`. Is this something we should make user configurable?

Reply:
The two should be the same: a chunk should end on an element-ending delimiter.

Reply:
OK great that was my feeling but was worried I wasn't accounting for some extra case